internal/chat/model.go

```diff
@@ -0,0 +1,58 @@
+package chat
+
+import (
+	"sync"
+
+	"github.com/iggyster/lets-go-chat/internal/user"
+)
+
+var (
+	c    IChat
+	once sync.Once
+)
+
+type IChat interface {
```

Comment:
Start interface name from I - it isn't Go style.
https://go.dev/doc/effective_go#names